test/schema_test.ts

```diff
@@ -371,6 +371,38 @@ describe('Schema', () => {
       });
     });
 
+    it('works with predefined identiry', () => {
+      const thingSchema = makeObjectSchema({
+        identity: {
+          name: 'Thing',
+        },
+        properties: {
+          name: {type: ValueType.String, required: true},
+          id: {type: ValueType.Number, required: true},
+          other: {type: ValueType.Boolean},
+        },
+        displayProperty: 'name',
+        idProperty: 'id',
+      });
+
+      const thingReferenceSchema = schema.makeReferenceSchemaFromObjectSchema(thingSchema);
```

Comment:
(I noticed we didn't have any test for a schema that still sets its own identity)